pkg/cmd/cli/cmd/edit.go

```diff
@@ -74,66 +103,119 @@ func NewCmdEdit(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Com
 		},
 	}
 	usage := "Filename, directory, or URL to file to use to edit the resource"
-	kubectl.AddBoundJsonFilenameFlag(cmd, &filenames, usage)
+	kubectl.AddBoundJsonFilenameFlag(cmd, &options.filenames, usage)
 	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: yaml|json.")
 	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+
 	return cmd
 }
 
-// RunEdit contains all the necessary functionality for the OpenShift cli edit command
-func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, filenames util.StringList) error {
-	var printer kubectl.ResourcePrinter
-	var ext string
-	switch format := cmdutil.GetFlagString(cmd, "output"); format {
+//Method Complete() completes the struct variables
+func (o *EditOptions) Complete() error {
+	switch format := cmdutil.GetFlagString(o.cmd, "output"); format {
 	case "json":
-		printer = &kubectl.JSONPrinter{}
-		ext = ".json"
+		o.printer = &kubectl.JSONPrinter{}
+		o.ext = ".json"
 	case "yaml":
-		printer = &kubectl.YAMLPrinter{}
-		ext = ".yaml"
+		o.printer = &kubectl.YAMLPrinter{}
+		o.ext = ".yaml"
 	default:
-		return cmdutil.UsageError(cmd, "The flag 'output' must be one of yaml|json")
+		return cmdutil.UsageError(o.cmd, "The flag 'output' must be one of yaml|json")
 	}
 
-	cmdNamespace, explicit, err := f.DefaultNamespace()
+	cmdNamespace, explicit, err := o.f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
+	o.namespace = cmdNamespace
 
-	mapper, typer := f.Object()
-	rmap := &resource.Mapper{
+	mapper, typer := o.f.Object()
+	o.rmap = &resource.Mapper{
 		ObjectTyper:  typer,
 		RESTMapper:   mapper,
-		ClientMapper: f.ClientMapperForCommand(),
+		ClientMapper: o.f.ClientMapperForCommand(),
 	}
 
-	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
-		NamespaceParam(cmdNamespace).DefaultNamespace().
-		FilenameParam(explicit, filenames...).
+	o.builder = resource.NewBuilder(mapper, typer, o.f.ClientMapperForCommand()).
+		NamespaceParam(o.namespace).DefaultNamespace().
+		FilenameParam(explicit, o.filenames...).
 		//SelectorParam(selector).
-		ResourceTypeOrNameArgs(true, args...).
+		ResourceTypeOrNameArgs(true, o.args...).
 		Latest()
+
+	clientConfig, err := o.f.ClientConfig()
 	if err != nil {
 		return err
 	}
 
-	clientConfig, err := f.ClientConfig()
+	r := o.builder.Flatten().Do()
+	o.infos, err = r.Infos()
 	if err != nil {
 		return err
 	}
 
-	r := b.Flatten().Do()
-	infos, err := r.Infos()
-	if err != nil {
-		return err
+	o.version = cmdutil.OutputVersion(o.cmd, clientConfig.Version)
+	o.results = &editResults{}
+	return nil
+}
+
+//Method Validate() performs validation and returns errors
+func (o *EditOptions) Validate() error {
+	if o.out == nil {
```

Comment:
Checking `o.out` seems a bit suspicious here, shouldn't it be defined within `Complete`?
[Here](https://github.com/openshift/origin/blob/255307e708defdbdb66fc421cc3d45cce375a209/pkg/cmd/cli/cmd/deploy.go#L130) is where it is set in case of `oc deploy` command, and it is not checked in `Validate`. I imagine only user-provided data needs to go through validation?